x-pack/plugins/files/public/components/upload_file/upload_file.component.tsx

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import {
+  EuiIcon,
+  EuiText,
+  EuiButton,
+  EuiSpacer,
+  useEuiTheme,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiFilePicker,
+  EuiButtonEmpty,
+  type EuiFilePickerProps,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+import { euiThemeVars } from '@kbn/ui-theme';
+import { css } from '@emotion/react';
+import { i18nTexts } from './i18n_texts';
+
+export interface Props
+  extends Omit<EuiFilePickerProps, 'onChange' | 'value' | 'initialPromptText' | 'disabled'> {
+  onChange: (files: File[]) => void;
+  onUpload: () => void;
+  onClear: () => void;
+  onCancel: () => void;
+  label?: string;
+  errorMessage?: string;
+  accept?: string;
+  done?: boolean;
+  ready?: boolean;
+  uploading?: boolean;
+  immediate?: boolean;
+  retry?: boolean;
+  allowClear?: boolean;
+  initialFilePromptText?: string;
+}
+
+const { euiFormMaxWidth, euiButtonHeight, euiButtonHeightSmall } = euiThemeVars;
+
+export const UploadFile = React.forwardRef<EuiFilePicker, Props>((props, ref) => {
+  const {
+    done,
+    label,
+    ready,
+    retry,
+    style,
+    accept,
+    onClear,
+    onCancel,
+    onChange,
+    onUpload,
+    className,
+    immediate,
+    uploading,
+    isInvalid,
+    compressed,
+    errorMessage,
+    allowClear = false,
+    initialFilePromptText,
+    ...rest
+  } = props;
+
+  const { euiTheme } = useEuiTheme();
+  const { size } = euiTheme;
+  const id = useGeneratedHtmlId({ prefix: 'filesUploadFile' });
+  const errorId = `${id}_error`;
+
+  const controllButton = (): JSX.Element | null => {
```

Comment:
Would be nice to make this `controlButton()` function a connected `<ControlButton>` component, something like this:

```tsx
export const ControlButton: React.FC<Props> = () => {
  const uploadState = useUploadState();
  const uploading = useBehaviorSubject(uploadState.uploading$)
  const uploads = useBehaviorSubject(uploadState.uploads$)

  if (uploading) return <CancelButton />;
  if (uploads.length) return <RetryButton />;
  if (!done && !immediate) return <UploadButton />
  if (done) return <SuccessMessage />;
  return null;
};
```

Reply:
Maybe success message should not be part of this component.